### PR TITLE
Adding acoustic wave on sphere balance law example

### PR DESCRIPTION
**Pull request by jkozdon**
This adds the acoustic wave test case of [Tomita and Satoh (2004)](https://doi.org/10.1016/j.fluiddyn.2004.03.003) to the balance law examples.

This also includes the skew-form metric treatment from #262

@tapios and @charleskawczynski  these use the moist thermodynamics routines so it would be great to get your review on the file `examples/DGmethods/ex_003_acoustic_wave.jl` concerning the use of this functionality

### CODE REVIEW

**Comment by jkozdon** [comment]
FYI @lcw

**Comment by jkozdon** [comment]
bors try

File: examples/DGmethods/ex_003_acoustic_wave.jl [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+
+    ρ = ρ_ref + dρ
+    ρe = ρe_ref + dρe
+    e = ρe / ρ
+
+    ## compute the velocity
+    u, v, w = ρu / ρ, ρv / ρ, ρw / ρ
+
+    ## internal energy
+    e_int = e - (u^2 + v^2 + w^2)/2 - ϕ
```
Comment (tapios) [inline examples/DGmethods/ex_003_acoustic_wave.jl]: This is fine. But we should probably add a function to calculate internal energy from total in `MoistThermodynamics`. @charleskawczynski ?

File: examples/DGmethods/ex_003_acoustic_wave.jl [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+    ρ = ρ_ref + dρ
+    e = (ρe_ref + dρe) / ρ
+
+    ## velocity field
+    u, v, w = ρu / ρ, ρv / ρ, ρw / ρ
+
+    ## internal energy
+    e_int = e - (u^2 + v^2 + w^2)/2 - ϕ
+
+    ## compute the pressure
```
Comment (tapios) [inline examples/DGmethods/ex_003_acoustic_wave.jl]: typo: temperature in comment

**Comment by tapios** [comment]
This looks good to me as far as thermodynamics are concerned. As I mentioned in a comment, we should probably add a function to compute internal energy from total energy and state in `MoistThermodynamics`. @charleskawczynski ?

**Comment by jkozdon** [comment]
bor try

**Comment by jkozdon** [comment]
bors try

**Comment by lcw** [comment]
bors try

**Comment by lcw** [comment]
@simonbyrne The CI seems to be timing out.  What do we need to do to get this example merged?

**Comment by jkozdon** [comment]
bump?

File: examples/DGmethods/ex_003_acoustic_wave.jl (hunk 1 above)
Comment (charleskawczynski) [reply examples/DGmethods/ex_003_acoustic_wave.jl]: Yes, I'll add a wrapper function for MoistThermo.

**Pull request merged**